tests/jeos/firstrun.pm

```diff
@@ -44,9 +49,34 @@ sub verify_user_info {
     assert_script_run("ldd --help | grep '^" . $lang_data{$proglang} . "'");
 }
 
+sub verify_mounts {
+    my $expected    = {mount_type => (expect_mount_by_uuid) ? 'UUID=' : 'LABEL='};
+    my $fstab_count = 0;
+
+    # get UUID of rootfs partition
+    (script_output('cat /proc/cmdline') =~ /root=UUID=(?<rootfs>\w+(\-\w+)+)\b/) or
+      die 'No root partition found among bootparameters!';
+    record_info("rootfs", $+{rootfs});
+
+    # expected format (UUID|LABEL)=($uuid|ROOT|EFI)
+    foreach my $entry (split(/\n/, script_output('cat /etc/fstab'))) {
+        my $uuid_or_label = (expect_mount_by_uuid) ? $+{rootfs} : ($entry =~ m[\s/boot/efi\s]) ? 'EFI' : 'ROOT';
+        print "TESTa  " . Dumper($uuid_or_label) . "\n";
+        ($entry =~ qr/^$expected->{mount_type}$uuid_or_label\b/) or next;
+        $fstab_count++;
+    }
+
+    my $findmnt_count = scalar grep { /\s$expected->{mount_type}.*\stranslated\sto\s/ } split(/\n/, script_output('findmnt --verbose --verify'));
+
+    diag "\nMount entries in fstab *** $fstab_count ***\nFound by findmnt *** $findmnt_count ***\n";
+    (($findmnt_count == $fstab_count) && ($findmnt_count != 0) && ($fstab_count != 0)) or
+      die "Expected mounts do not fit, found mount entries ($fstab_count) in fstab, found in ($findmnt_count) findmnt\n";
```

Comment:
Doesn't `findmnt --verify` do that already?

Reply:
Yup, it checks fstab's parsability and usability, which is in fact the point. I meant to double-check the mounts in some way, so parsing fstab is obviously the primary option and nothing better did not come to my mind than findmnt. Any ideas? :)

Reply:
Why double-check it? I would hope that findmnt knows what it's doing...

The current hand-made parser would break for things such as comments, for example

Reply:
True, we can trust `findmnt`. I went the way around. I have done the `fstab` parsing manually and then verify against `findmnt`.

Reply:
Yes, but why? It's just more complexity

Reply:
it brought some value to verify whether I am getting correct results from fstab parsing. But from the point of the test case, it does not bring any additional value as you pointed out.